plots/plots_validation.py

```diff
@@ -0,0 +1,644 @@
+
+
+def run(country_code, horizon):
+
+    # Load Ember data
+    ember_data = load_ember_data()
+
+    # Load PyPSA network
+    # Note: you can also load the network directly using """network = pypsa.Network(network_path)"""
+    # network = load_pypsa_network(scenario_folder)
+    network = load_network(get_solved_network_path(
+        country_code, horizon, PYPSA_RESULTS_DIR))
+
+    # Load EIA data
+    EIA_demand_path = os.path.join(DATA_DIR, "EIA_demands.csv")
+    EIA_installed_capacities_path = os.path.join(
+        DATA_DIR, "EIA_installed_capacities.csv")
+    EIA_generation_path = os.path.join(
+        DATA_DIR, "EIA_electricity_generation.csv")
+
+    three_country_code = convert_two_country_code_to_three(country_code)
+    build_directory(PLOTS_DIR, just_parent_directory=True)
+
+    demand_ember = get_demand_ember(ember_data, three_country_code, horizon)
+    pypsa_demand = get_demand_pypsa(network)
+
+    EIA_demand = get_data_EIA(EIA_demand_path, country_code, horizon)
+    EIA_demand = EIA_demand.iloc[0, 1]
+
+    installed_capacity_ember = get_installed_capacity_ember(
+        ember_data, three_country_code, horizon)
+    pypsa_capacity = get_installed_capacity_pypsa(network)
+
+    EIA_inst_capacities = get_data_EIA(
+        EIA_installed_capacities_path, country_code, horizon)
+    EIA_inst_capacities = preprocess_eia_data(EIA_inst_capacities)
+
+    installed_capacity_df = pd.concat(
+        [installed_capacity_ember, pypsa_capacity, EIA_inst_capacities], axis=1).fillna(0)
+
+    generation_data_ember = get_generation_capacity_ember(
+        ember_data, three_country_code, horizon)
+    pypsa_generation = get_generation_capacity_pypsa(network)
+
+    EIA_generation = get_data_EIA(EIA_generation_path, country_code, horizon)
+    EIA_generation = preprocess_eia_data(EIA_generation)
+
+    generation_df = pd.concat(
+        [generation_data_ember, pypsa_generation, EIA_generation], axis=1).fillna(0)
+
+    generation_data_ember_ = get_generation_capacity_ember_detail(
+        ember_data, three_country_code, horizon)
+    EIA_generation_ = get_data_EIA(EIA_generation_path, country_code, horizon)
+    EIA_generation_ = preprocess_eia_data_generation(EIA_generation_)
+    pypsa_generation_ = get_generation_capacity_pypsa_detail(network)
+
+    generation_df_ = pd.concat(
+        [generation_data_ember_, pypsa_generation_, EIA_generation_], axis=1).fillna(0)
+
+    plot_demand_validation(demand_ember, pypsa_demand,
+                           EIA_demand, country_code, horizon)
+    plot_capacity_validation(installed_capacity_df, horizon, country_code)
+    plot_generation_validation(generation_df, horizon, country_code)
+    plot_detail_generation_validation(generation_df_, horizon, country_code)
+
+
+if __name__ == "__main__":
+    if "snakemake" not in globals():
+        snakemake = mock_snakemake(
+            "plot_validation",
+            countries="US",
+            planning_horizon="2021",
```

Comment:
Here the name of the rule was given, Also horizon was placed according to data that we have (2021 simulations).